File: src/client/FantomasTools/FSharpTokens/Encoders.fs
module FantomasTools.Client.FSharpTokens.Encoders

open FantomasTools.Client.FSharpTokens.Model
open Thoth.Json

let encodeGetTokensRequest (value: FSharpTokens.Shared.GetTokensRequest): JsonValue =
    Encode.object [ "defines",
                    (value.Defines
                     |> List.map Encode.string
                     |> List.toArray
                     |> Encode.array)
                    "sourceCode", Encode.string value.SourceCode ]

let encodeUrlModel code model: JsonValue =
    Encode.object [ "defines", Encode.string model.Defines
                    "code", Encode.string code ]

File: _dist_/bin/FSharpTokens/Encoders.js
import { object } from "../.fable/Thoth.Json.5.0.0/Encode.fs.js";
import { map, toArray } from "../.fable/fable-library.3.1.15/List.js";

export function encodeGetTokensRequest(value) {
    return object([["defines", toArray(map((value_1) => value_1, value.Defines))], ["sourceCode", value.SourceCode]]);
}

export function encodeUrlModel(code, model) {
    return object([["defines", model.Defines], ["code", code]]);
}

